openshift/jag-csrs.yml

```diff
@@ -0,0 +1,164 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: ${APP_NAME}
+  description: Deployment template for the JAG-csrs
+  
+labels:
+  app: ${APP_NAME}
+  app.kubernetes.io/name: ${COMPONENT_NAME}
+  app.kubernetes.io/part-of: ${APP_NAME}
+
+  
+parameters:
+
+- name: APP_NAME
+  description: The application name that each component is part of
+  required: true
+  value: jag-csrs-frontend
+
+- name: SPLUNK_CONFIG_NAME
+  description: The application name that each component is part of
+  required: true
+  value: blahblah
+
+- name: ROUTE_PREFIX
+  description: The application name that each component is part of
+  required: true
+  value: jag-csrs-frontend
+
+- name: COMPONENT_NAME
+  description: The component name
+  required: true
+  value: jag-csrs-frontend
+  
+- name: OC_NAMESPACE
+  description: The OpenShift namespace prefix
+  required: true
+  # Below value must be changed as per gold cluster oc nameplate
+  value: blahblah
+  
+- name: OC_ENV
+  description: The OpenShift environment, ie dev, test or prod
+  required: true
+  value: dev
+
+- name: DNS
+  description: The application name that each component is part of
+  required: false
+  # Below value must be changed as per gold cluster certificate/DNS
+  value: apps.silver.devops.gov.bc.ca
+
+- name: BCEID_URL
+  description: BCEID_URL for dev, test or prod
+  required: true
+  value: https://www.development.bceid.ca/
+  
+objects:
+
+### JAG-csrs Deployment Config ###
+- apiVersion: apps.openshift.io/v1
+  kind: DeploymentConfig
+  metadata:
+    name: ${COMPONENT_NAME}
+    namespace: ${OC_NAMESPACE}-${OC_ENV}
+      
+  spec:
+    completionDeadlineSeconds: 1200
+    replicas: 1
+    selector:
+      name: ${COMPONENT_NAME}
+    strategy:
+      type: Rolling
+      maxSurge: 50%
+      maxUnavailable: 0
+    template:
+      metadata:
+        labels:
+          name: ${COMPONENT_NAME}
+      spec:
```

Comment:
Add label like this:

```yaml
  labels:
    app: csrs-portal
    app.kubernetes.io/component: frontend
    app.kubernetes.io/instance: csrs-portal
    app.kubernetes.io/name: csrs-portal
    app.kubernetes.io/part-of: csrs
    app.openshift.io/runtime: angularjs
```

it will help in the topology view in Openshift, for example in TCO

![image](https://user-images.githubusercontent.com/1844480/143278628-0097e93b-6385-43da-b616-1d371aad1e15.png)

Reply:
The labels are already there 

![image](https://user-images.githubusercontent.com/18105535/143317137-1aee24a4-1b86-42eb-91da-5c76cb667b2f.png)

![image](https://user-images.githubusercontent.com/18105535/143317366-5d4c6dc5-ada1-4203-a898-f323867a99fa.png)


And that is inherited from 
![image](https://user-images.githubusercontent.com/18105535/143317263-ab9283d8-6927-4d7e-9f6e-0e04efc6ece2.png)

What could be useful but missing is the runtime. I am going to add that. Thanks!